docs/adr/adr-003-march2022-testnet.md

```diff
@@ -0,0 +1,154 @@
+# ADR #003: March 2022 Testnet Celestia Node
+
+<hr style="border:3px solid gray"> </hr>
+
+## Authors
+
+@renaynay @Wondertan
+
+## Changelog
+
+* 2021-11-25: initial draft
+
+<hr style="border:2px solid gray"> </hr>
+
+## Legend
+
+### Celestia DA Network
+
+Refers to the data availability "halo" network created around the Core network.
+
+### **Bridge Node**
+
+A **bridge** node is a **full** node that is connected to a Celestia Core node via RPC. It receives either a remote
+address from a running Core node or it can run a Core node as an embedded process, but the critical difference is that
+instead of constructing blocks via sampling the network for shares, it receives headers and blocks directly from its 
+trusted Core node, validating blocks and producing `ExtendedHeader`s to broadcast to the Celestia DA network.
+
+### **Full Node**
+
+A **full** node is the same thing as a **light** node, but instead of performing `LightAvailability` (the process of 
+DASing to verify a header is legitimate), it performs `FullAvailability` which samples the network for shares in order 
+to fully reconstruct the block and store it, serving shares to the rest of the network.
+
+### **Light Node**
+
+A **light** node listens for `ExtendedHeader`s from the DA network and performs DAS on the received headers.
+
+<hr style="border:2px solid gray"> </hr>
+
+## Context
+
+This ADR describes a design for the March 2022 Celestia Testnet that we decided at the Berlin 2021 offsite. Now that
+we have a basic scaffolding and structure for a celestia node, the focus of the next engineering sprint is to continue 
+refactoring and improving this structure to include more features (defined later in this document).
+
+
+<hr style="border:2px solid gray"> </hr>
+
+## Decision
+
+## New Features
+
+### New node type definitions
+* Introduce a standalone **full** node and rename current full node implementation to **bridge** node. 
+* Remove **dev** as a node type and make it a flag on every available node type.
+
+### Introduce bad encoding fraud proofs
+Bad encoding fraud proofs will be generated by **full** nodes inside of `ShareService`, upon reconstructing a block
+via the sampling process. 
+
+If fraud is detected, the **full** node will generate the proof and propagate it to the `FraudSub` gossip network. 
+If no fraud is detected, _______________________@TODO_________________________?
+
+If a fraud proof is received via `FraudSub`, **full** nodes will perform the verification themselves to check to ensure
+the fraud proof is valid. 
+If the fraud proof is valid, the node should immediately halt all operations.
+If it is invalid, __________________@TODO____________? (Do nothing?).
```

Comment:
cc @liamsi @Wondertan 

My idea here: doesn't libp2p have some sort of built-in peer reputation management system where you can mark a certain peer as less trustworthy than others based on "bad behaviour"? **This isn't necessary to implement for the next testnet iteration**, but I think it would be nice to tag certain peers as misbehaving if they, for any reason, propagate incorrect bad encoding proofs to the network since the behaviour for a node upon receiving a valid fraud proof is to halt all operations.

Reply:
> My idea here: doesn't libp2p have some sort of built-in peer reputation management system where you can mark a certain peer as less trustworthy than others based on "bad behaviour"? 

Gossipsub (v1.1) has something in this direction but I am not very familiar with the details yet:
https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#peer-scoring